src/test/crimson/seastar_runner.h

```diff
@@ -20,37 +20,50 @@ struct SeastarRunner {
 
   std::thread thread;
 
+  bool app_started = false;
+  bool begin_signaled = false;
+
   SeastarRunner() :
     begin_fd{seastar::file_desc::eventfd(0, 0)} {}
 
   ~SeastarRunner() {}
 
-  void init(int argc, char **argv)
+  int init(int argc, char **argv)
   {
     thread = std::thread([argc, argv, this] { reactor(argc, argv); });
     eventfd_t result = 0;
     if (int r = ::eventfd_read(begin_fd.get(), &result); r < 0) {
       std::cerr << "unable to eventfd_read():" << errno << std::endl;
-      throw std::runtime_error("Cannot start seastar");
+      return r;
+    }
+    assert(begin_signaled == true);
+    if (!app_started) {
+      return 1;
     }
+    return 0;
   }
   
   void stop()
   {
-    run([this] {
-      on_end->write_side().signal(1);
-      return seastar::now();
-    });
+    if (app_started) {
```

Comment:
```suggestion
    if (on_end) {
      on_end->write_side().signal(1);
      // ...
    }
```

Reply:
Yeah, `app_started` is unnecessary and can be replaced by checking `on_end`.